activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -62,7 +62,7 @@ module DatabaseTasks
       LOCAL_HOSTS = ["127.0.0.1", "localhost"]
 
       def check_protected_environments!
-        unless ENV["DISABLE_DATABASE_ENVIRONMENT_CHECK"]
+        unless ENV["DISABLE_DATABASE_ENVIRONMENT_CHECK"] == "1"
```

Comment:
@NielsKSchjoedt I'd say that `DISABLE_DATABASE_ENVIRONMENT_CHECK` shouldn't be set at all if you desire a falsy response than checking specifically for `1`. Usually, for boolean environment variables, their presence is all you need to check, not the value they're assigned to.